src/context/DataContext.jsx

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useState, useEffect } from "react";
+
+const DataContext = createContext();
+
+const useData = () => useContext(DataContext);
+
+const DataProvider = ({ children }) => {
+  const [taskList, setTaskList] = useState(
+    JSON.parse(localStorage.getItem("tasks")) ?? []
```

Comment:
Nice use of nullish coalescing operator for handling undefined return from JSON.parse

Reply:
Thanks for the suggestion will change the colors in the end while doing style optimizations.